taworks/taform/views.py

```diff
@@ -19,6 +19,120 @@
 from access_tokens import scope, tokens
 import uuid
 import os.path
+from django.core import mail
+from threading import Thread
+
+# This is to provide annotation for methods that need a separate thread
+def postpone(function):
+  def decorator(*args, **kwargs):
+    t = Thread(target = function, args=args, kwargs=kwargs)
+    t.daemon = True
+    t.start()
+  return decorator
+
+def ranking_status(request):
+    if not request.user.is_authenticated:
+        return redirect('login')
+    elif 'Upload' in request.POST:
+        email_ranking_links(request.POST['email'])
+        return render(request, 'taform/ranking_status.html', 
+            {'success': 'Ranking email links have been sent.', 'sent': True })
+    return render(request, 'taform/ranking_status.html', {'sent': False})
+
+@postpone
+def email_ranking_links(report_email = None):
+    connection = mail.get_connection()
+    connection.open()
+
+    courses = models.Course.objects.all()
+    email = []
+
+    for i, course in enumerate(courses):
+        tmp = mail.EmailMessage(
+            'TA Ranking Form for {course_name}'.format(
+                course_name = course.course_name),
+            """
+            <div>Dear {instructor},</div>
+            <br/>
+            <div>Please click on the link below and follow the instructions to
+            complete and submit your rankings for the course 
+            ({subject} {id}-{section})</div>
+            <br/>
+            <div>Link to Ranking Page: 
+            https://team4.uwaterloo.ca/taform/instructor/{url}</div>
+            <br/>
+            <div>Regards,</div>
+            <div>Associate Chair for Undergraduate Studies, Management Sciences
+            </div>
+            <br/>
+            <div>*Note: If you are the Instructor for more than one course in the 
+            upcoming term, you will receive an email for each course. Important: 
+            Links are specific to each class.
+            You may also need to be on campus wifi or vpn if you are remote.</div>
```

Comment:
can we make this bold? Just to make it stand out more -- just my opinion but good either way!

Reply:
Done